src/Microsoft.IdentityModel.Tokens/TokenValidationParameters.cs

```diff
@@ -762,5 +784,107 @@ public string RoleClaimType
+        /// recursion or deeply nested actor tokens.</para>
+        /// <para>In most scenarios, users don't need to set this property as it's managed internally 
+        /// by the token validation and creation process.</para>
+        /// </remarks>
+        public int ActorChainDepth
+        {
+            get => _actorClainDepth;
+            set
+            {
+                _actorClainDepth = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the delegate that will be used to convert the 'act' claim JSON into a ClaimsIdentity.
+        /// <para>This delegate is invoked during token validation when an actor claim is encountered in a token.</para>
+        /// <para>The delegate receives a <see cref="System.Text.Json.JsonElement"/> representing the actor claim
+        /// and should return a <see cref="ClaimsIdentity"/> that represents the actor.</para>
+        /// </summary>
+        /// <remarks>
+        /// <para>When this delegate is provided, it replaces the default actor claim processing logic.</para>
+        /// <para>This is useful for custom actor claim formats or when special processing is needed for the actor claims.</para>
+        /// <para>The delegate can also handle nested actors by recursively creating actor identities and setting the Actor property.</para>
+        /// <code>
+        /// validationParameters.ActClaimRetrieverDelegate = (JsonElement element,TokenValidationParameters tokenValidationParameters) => {
+        ///     var identity = new ClaimsIdentity("CustomActor");
+        ///     // Extract claims from the JsonElement
+        ///     if (element.TryGetProperty("sub", out var sub))
+        ///         identity.AddClaim(new Claim("sub", sub.GetString()));
+        ///     return identity;
+        /// };
+        /// </code>
+        /// </remarks>
+        public ActClaimRetrieverDelegate ActClaimRetrieverDelegate { get; set; }
+
+        /// <summary>
+        /// Gets or sets the maximum depth allowed when processing nested actor tokens.
+        /// <para>This prevents excessive recursion when handling deeply nested actor tokens.</para>
+        /// <para>The value must be at least 0. Value 0 would mean that no actor token nesting is allowed.</para>
+        /// <para>The maximum allowed value is 4 to prevent security issues with excessively deep actor chains.</para>
+        /// </summary>
+        /// <remarks>
+        /// <para>Default value is 4.</para>
+        /// <para>During token validation and creation, an exception will be thrown if the actor nesting exceeds this limit.</para>
+        /// <para>This limit applies to both token creation and validation processes.</para>
+        /// </remarks>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown if the value is less than 0 or greater than 4.</exception>
+        public int MaxActorChainLength
```

Comment:
From the previous discussions, was there a request to make `MaxActorChainLength` and `ActorChainDepth` be setteable by the user?